json-format.md

```diff
@@ -141,22 +141,55 @@ For any other type, the implementation MUST translate the data value into a
 [JSON value][json-value], and use the member name `data` to store it inside the
 JSON object.
 
+If the implementation determines that the type of data is NOT `Binary`, it MUST
+next determine whether the `datacontenttype` attribute, if present, indicates a
+JSON-oriented content type. A JSON-oriented content type is one whose subtype
+is `json` or has a JSON format; that is, a MIME type of the form `*/json` or
+`*/*+json`. If the `datacontenttype` attribute is absent, it is treated as if it
```

Comment:
That sounds reasonable. I'm running under the assumption that even if we were to explicitly set an absent `datacontenttype` to `"application/json"`, it would not change the meaning of the event. However, maybe we should be more strict. I'll see what I can do.

Reply:
Actually, rereading this, I think this _could_ imply that an SDK should explicitly set `datacontenttype` to `"application/json"` if a JSON-formatted event came in without an explicit `datacontenttype` attribute. Otherwise, as I think you allude to later on, the application (and/or subsequent encoders) must have knowledge that (1) the event was decoded from JSON format, AND (2) therefore its "effective" content type is JSON.

Different SDKs may implement this differently. The Ruby SDK explicitly sets `datacontenttype` when it decodes a JSON-formatted event that omits the attribute. Another SDK might preserve the fact that the attribute was absent in the original event, but represent in some other way that the data should be interpreted as a JSON value. Either way, an SDK should have a way to make _explicit_ the fact that a missing attribute implies a JSON-typed data attribute, if for no other reason than to ensure that if the event is subsequently re-encoded in some other format or in binary mode, that can be done without loss of information. I'm not sure how or to what extent we'd need to say this, though. Ideas welcome.

Reply:
@duglin: A lot of this is really about how SDKs behave. I think trying to write the spec without making expected SDK behavior clear is going to lead to inconsistency. In a previous meeting we started talking a bit about how/where we want SDK behavior to be specified - I think it deserves more conversation.

In terms of how I think this particular one should work: I *think* I'd prefer to avoid inferring too much information in the CloudEvent returned by the SDK. It feels like that's taking it a step too far - to be honest, I wish we didn't have this defaulting in the spec at all, but it's too late to fix that. I'm sufficiently ambivalent to prefer consistency over dogma though - I'm happy to make the C# SDK follow the crowd.

Reply:
One area of confusion here was that this section is talking about serialization, whereas the implicit behavior probably should apply _only_ to deserialization. (In particular, the text in spec.md is specific to deserialization.) That was an error on my part. I'm removing it from here and moving it to the deserialization section.

Reply:
Pushed a new commit, removing the "implicit" behavior from serialization, instead discussing it only during deserialization.